metricq_tools/summary.py

```diff
@@ -0,0 +1,294 @@
+        self.print_stats = print_stats
+        self.print_data = print_data
+
+        self.timestamps = dict[str, list]()
+        self.last_timestamp = dict[str]()
+        self.intervals = dict[str, list]()
+        self.values = dict[str, list]()
+
+        for metric in metrics:
+            self.timestamps[metric] = list()
+            self.last_timestamp[metric] = None
+            self.intervals[metric] = list()
+            self.values[metric] = list()
+
+    async def add_data(self, metric: str, timestamp: metricq.Timestamp, value: float):
+        if self.print_data:
+            click.echo(click.style("{}: {}".format(timestamp, value), fg="bright_blue"))
+
+        self.timestamps[metric].append(timestamp.posix)
+        if self.last_timestamp[metric]:
+            self.intervals[metric].append(timestamp.posix - self.last_timestamp[metric])
+        self.last_timestamp[metric] = timestamp.posix
+        self.values[metric].append(value)
+
+    def _print_histogram(self, values):
+        counts, bin_edges = np.histogram(values, bins="doane")
+        fig = tpl.figure()
+        labels = [
+            "[{:#.6g} - {:#.6g})".format(bin_edges[k], bin_edges[k + 1])
+            for k in range(len(bin_edges) - 2)
+        ]
+        labels.append(
+            "[{:#.6g} - {:#.6g}]".format(
+                bin_edges[len(bin_edges) - 2], bin_edges[len(bin_edges) - 1]
+            )
+        )
+        fig.barh(counts, labels=labels)
+        fig.show()
+
+    def _print_intervals_histogram(self, intervals):
+        click.echo(
+            click.style(
+                "Distribution of the duration between consecutive data points in seconds",
+                fg="yellow",
+            )
+        )
+        click.echo()
+
+        self._print_histogram(intervals)
+
+        click.echo()
+        click.echo()
+
+    def _print_values_histogram(self, values):
+        click.echo(
+            click.style("Distribution of the values of the data points", fg="yellow")
+        )
+        click.echo()
+
+        self._print_histogram(values)
+
+        click.echo()
+        click.echo()
+
+    def print(self):
+        for metric in self._metrics:
+
+            if self.values[metric]:
+                click.echo()
+                click.echo(click.style(f"Stats of metric {metric!r}:", fg="green"))
+                click.echo()
+
+                if self.print_intervals and self.last_timestamp[metric]:
+                    self._print_intervals_histogram(self.intervals[metric])
+
+                if self.print_values:
+                    self._print_values_histogram(self.values[metric])
+            else:
+                click.echo()
+                click.echo(
+                    click.style(f"No Data for metric {metric!r} received!", fg="red")
+                )
+                click.echo()
+
+        if self.print_stats:
+            self._print_statistics()
+
+    def _print_statistics(self):
```

Comment:
Would it be possible to transpose the table? I think its more natural to have headers for columns `Metric | Minimum | Maximum | ...` and then print stats for one metric per row.

This probably also scales better with a large number of metrics, imagine someone does a `metricq-summary -m foo.a01 ... -m foo.a50 ...`.